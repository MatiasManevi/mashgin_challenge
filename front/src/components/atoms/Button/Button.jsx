import React from 'react';
import clsx from 'clsx';

import './Button.scss';

const Button = ({ onClick, children, color = "primary" }) => {
	return <button className={clsx(color)} onClick={onClick}>{children}</button>;
};

export default Button;
