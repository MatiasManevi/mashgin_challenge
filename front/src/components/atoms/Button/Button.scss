@import 'assets/scss/variables';

button {
	cursor: pointer;
	color: $color-neutral;
	background-color: $color-primary;
	border: none;
	user-select: none;
	padding: $space-8;
	border-radius: $space-8;
	font-weight: 700;
	font-size: $font-size-18;
	transition: all .5s ease;

	&.secondary {
		background-color: $color-secondary;
		color: $color-white;
	}

	&.neutral {
		background-color: $color-neutral;
		color: $color-white;
	}

	&:hover {
		filter: brightness(.95);
	}
}
