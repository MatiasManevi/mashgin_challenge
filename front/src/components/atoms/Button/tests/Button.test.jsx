import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('Button', () => {
	const clickHandler = jest.fn();

	it('should match snapshot', () => {
		const wrapper = shallow(<Button onClick={clickHandler} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call clickHandler when clicking', () => {
		const wrapper = shallow(<Button onClick={clickHandler} />);
		wrapper.simulate('click');
		expect(clickHandler).toHaveBeenCalledTimes(1);
	});
});
