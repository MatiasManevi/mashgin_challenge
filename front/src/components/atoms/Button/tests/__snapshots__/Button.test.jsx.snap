// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="primary"
  disabled={false}
  onClick={[MockFunction]}
/>
`;
